content/en/blog/_posts/2022-08-04-cgroupv2-ga.md

```diff
@@ -0,0 +1,95 @@
+
+Some Kubernetes features exclusively use cgroup v2 for enhanced resource management and isolation. For example, the [MemoryQoS feature](https://kubernetes.io/blog/2021/11/26/qos-memory-resources/) improves memory QoS and relies on cgroup v2 primitives to enable it. New resource management features in kubelet will also take advantage of the new cgroup v2 features moving forward.
+
+## How do I use cgroup v2?
+
+As you update your cluster and nodes to a newer Linux distribution, you may be already using cgroup v2!
+
+The recommended way to use cgroup v2 is to use a Linux distribution that enables and uses cgroup v2 by default, for example one of the many distributions below:
+
+* Container Optimized OS (since M97)
+* Ubuntu (since 21.10)
+* Debian GNU/Linux (since Debian 11 Bullseye)
+* Fedora (since 31)
+* Arch Linux (since April 2021)
+* RHEL and RHEL-like distributions (since 9)
+
+To check if your distribution uses cgroup v2 by default, you can follow the instructions [here](/docs/concepts/architecture/cgroups/#check-cgroup-version) or consult your OS distribution documentation.
+
+If you're using a managed kubernetes offering, consult with your provider to determine how they're adopting cgroup v2.
+
+Using cgroup v2 with kubernetes has the following requirements:
+
+* OS distribution enables cgroup v2 on kernel version 5.8 or later
+* Container runtime supports cgroup v2. For example:
+  * [containerd](https://containerd.io/) v1.4 or later
+  * [cri-o](https://cri-o.io/) v1.20 or later
+* The kubelet and the container runtime are configured to use the [systemd cgroup driver](/docs/setup/production-environment/container-runtimes#systemd-cgroup-driver)
+
+The kubelet and container runtime use a [cgroup driver](/docs/setup/production-environment/container-runtimes#cgroup-drivers) to set cgroup limits. When using cgroup v2, it's highly recommended that both kubelet and your container runtime (e.g. containerd or CRI-O) use [systemd cgroup driver](/docs/setup/production-environment/container-runtimes#systemd-cgroup-driver). This ensures that there is only a single cgroup manager on the system. Refer to the [systemd cgroup driver documentation](/docs/setup/production-environment/container-runtimes#systemd-cgroup-driver) about how to configure kubelet and the container runtime to use systemd as the cgroup driver.
+
+## Migrating to cgroup v2
+
+When you run kubelet on a Linux distribution that enables cgroup v2, kubelet should automatically perform accordingly without any additional configuration required.
+
+You should not find any noticeable difference in the user experience when switching to cgroup v2, unless users are accessing the cgroup file system directly, either on the node or from within the containers.
+
+It's important to note that the cgroup v2 uses a different API compared to cgroup v1, so if there any applications that read the cgroup file system, they should be updated to support cgroup v2. For example:
```

Comment:
```suggestion
In most cases, you won't see a difference in the user experience when you switch to using cgroup v2 unless
your users access the cgroup file system directly.

If you have applications that access the cgroup file system directly, either on the node or
from inside a container, you must update the applications to use the cgroup v2 API instead of the
cgroup v1 API.

Scenarios in which you might need to update to cgroup v2 include the following:
```